### DISCUSSION TITLE
how to clear Input type?

### DISCUSSION THREAD

**Discussion by marekyggdrasil** [Textualize/textual#3427] [Q&A] [answered]
How I compose

```python
yield Input(id='amount_input', placeholder='Amount')
```

then I attempt to clear it later as follows

```python
self.query_one('#amount_input').clear()
```

and I get

```
AttributeError: 'Input' object has no attribute 'clear'
```

so my question is, how to clear the `Input` type? I want to reset the value.

**Comment by marekyggdrasil** [comment, marked as answer]
Alright so I just did

```python
self.query_one('#amount_input').value = ''
```

seems to be ok...

**Comment by davep** [comment]
> I want to reset the value.

[The clue is in your question](https://textual.textualize.io/widgets/input/#reactive-attributes). ;-)

```python
self.query_one('#amount_input').value = ""
```

**Reply by TomJGooding** [reply]
I wonder though if `Input` _should_ have a clear method? There's many other widgets which have similar methods, including the new `TextArea`.

**Reply by willmcgugan** [reply]
It's not a bad idea. It's not unreasonable to assume its existance.
